SimPEG/electromagnetics/static/induced_polarization/simulation_2d.py

```diff
@@ -56,10 +65,48 @@ def fields(self, m):
             # re-uses the DC simulation's fields method
             self._f = super().fields(None)
 
+            if self.data_type == "apparent_chargeability":
+                if self.verbose is True:
+                    print(">> Data type is apparaent chargeability")
+
+                dc_voltage = self.dpred_dc(f=self._f)
+                dc_data = Data(self.survey, dc_voltage)
+                for src in self.survey.source_list:
+                    for rx in src.receiver_list:
+                        rx._dc_voltage = dc_data[src, rx]
+                        rx.data_type = self.data_type
+                        rx._Ps = {}
+
         self._pred = self.forward(m, f=self._f)
 
         return self._f
 
+    def dpred_dc(self, f=None):
```

Comment:
@jcapriot: may be there is a better way than this..

Reply:
part of me was thinking of just simplifying the ip simulation to just accept a dc_simulation as a parameter, which would then allow for arbitrary dimensionality and less repeated code.

Reply:
hmm ... that's a good idea. so take dc simulation as an input parameter of ip simulation, but when dc simulation is none, then generate one.